velox/expression/SimpleFunctionAdapter.h

```diff
@@ -278,7 +299,7 @@ class SimpleFunctionAdapter : public VectorFunction {
       uint64_t* nullBuffer = nullptr;
       auto* data = applyContext.result->mutableRawValues();
       auto writeResult = [&applyContext, &nullBuffer, &data](
-                             auto row, bool notNull, auto out) INLINE_LAMBDA {
+                             auto row, bool notNull, auto& out) INLINE_LAMBDA {
```

Comment:
this is called when out is primitive, its better to pass out by value.

it wont matter if this is inlined, but if not we are passing address always and dereferencing it instead of a value of potentially less than 8 bytes and no dereferencing.

Reply:
This leads to warnings because of line 325, if we return NULL directly out will not be initialized.

Since we compile by default with the -Werror,-Wsometimes-uninitialized flags, this causes compilation errors (I guess the other code paths add sufficient complexity that the compiler doesn't realize this).